rust-symcrypt/src/hmac.rs

```diff
@@ -0,0 +1,215 @@
+use core::ffi::c_void;
+use std::mem;
+use std::ptr;
+use symcrypt_sys;
+
+pub const SHA256_HMAC_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA256_RESULT_SIZE as usize;
+pub const SHA384_HMAC_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA384_RESULT_SIZE as usize;
+
+pub struct HmacSha256State {
+    state: symcrypt_sys::SYMCRYPT_HMAC_SHA256_STATE,
+    expanded_key: symcrypt_sys::SYMCRYPT_HMAC_SHA256_EXPANDED_KEY,
+}
+
+impl HmacSha256State {
+    pub fn new(key: &[u8]) -> Self {
+        let mut instance = HmacSha256State {
+            state: symcrypt_sys::SYMCRYPT_HMAC_SHA256_STATE::default(),
+            expanded_key: symcrypt_sys::SYMCRYPT_HMAC_SHA256_EXPANDED_KEY::default(),
+        };
+        unsafe {
+            symcrypt_sys::SymCryptHmacSha256ExpandKey(
+                &mut instance.expanded_key,
+                key.as_ptr(),
+                key.len() as symcrypt_sys::SIZE_T,
+            );
+            symcrypt_sys::SymCryptHmacSha256Init(&mut instance.state, &instance.expanded_key);
+        }
+        instance
+    }
+
+    pub fn append(&mut self, data: &[u8]) {
+        unsafe {
+            symcrypt_sys::SymCryptHmacSha256Append(
+                &mut self.state,
+                data.as_ptr(),
+                data.len() as symcrypt_sys::SIZE_T,
+            )
+        }
+    }
+
+    pub fn result(&mut self, result: &mut [u8; SHA256_HMAC_RESULT_SIZE]) {
+        unsafe {
+            symcrypt_sys::SymCryptHmacSha256Result(&mut self.state, result.as_mut_ptr());
+        }
+    }
+}
+
+impl Drop for HmacSha256State {
+    fn drop(&mut self) {
+        unsafe {
+            symcrypt_sys::SymCryptWipe(
+                ptr::addr_of_mut!(self.state) as *mut c_void,
+                mem::size_of_val(&mut self.state) as symcrypt_sys::SIZE_T,
+            );
+
+            symcrypt_sys::SymCryptWipe(
+                ptr::addr_of_mut!(self.expanded_key) as *mut c_void,
+                mem::size_of_val(&mut self.expanded_key) as symcrypt_sys::SIZE_T,
+            );
+        }
+    }
+}
+
+pub fn hmac256(key: &[u8], data: &[u8], result: &mut [u8; SHA256_HMAC_RESULT_SIZE]) {
```

Comment:
```suggestion
pub fn hmacsha256(key: &[u8], data: &[u8], result: &mut [u8; SHA256_HMAC_RESULT_SIZE]) {
```

Include SHA in the name; HMAC is a just a type of MAC, and the 256 is the size of the SHA2 digest. So `hmac256` is insufficiently clear.